python/TestHarness/schedulers/Scheduler.py

```diff
@@ -0,0 +1,539 @@
+    def notifySchedulers(self, tester):
+        return
+
+    # Return boolean on output ready to be read
+    def checkOutputReady(self, tester):
+        return True
+
+    # Call back to TestHarness to closeFiles in the event a derived scheduler wants to exit abnormally
+    def cleanup(self, exit_code=0):
+        self.harness.closeFiles(exit_code)
+
+    # return bool if we want to ignore prereq requirements
+    def skipPrereqs(self):
+        if self.options.ignored_caveats:
+            caveat_list = [x.lower() for x in self.options.ignored_caveats.split()]
+            if 'all' in self.options.ignored_caveats or 'prereq' in self.options.ignored_caveats:
+                return True
+        return False
+
+    # return a set of finished non-passing or will be skipped tests
+    def skippedTests(self, testers):
+        skipped_failed = set([])
+        for tester in testers:
+            if (tester.isFinished() and not tester.didPass()) \
+               or not tester.getRunnable(self.options):
+                skipped_failed.add(tester)
+        return skipped_failed
+
+    # Delete and return downstream nodes based on tester status and TestHarness options
+    # TODO: change the method name, as it doesn't always delete downstream tests
+    def deleteDownstreamTests(self, job_container):
+        with self.thread_lock:
+            failed_job_containers = set([])
+            tester = job_container.getTester()
+            job_dag = job_container.getDAG()
+            if (tester.isFinished() and not tester.didPass() and not tester.isSilent() and not self.skipPrereqs()) \
+                or (self.options.dry_run and not tester.isSilent()):
+
+                # Ask the DAG to delete and return the downstream jobs associated with this job
+                failed_job_containers.update(job_dag.delete_downstreams(job_container))
+
+            for failed_job in failed_job_containers:
+                tester = failed_job.getTester()
+                tester.setStatus('skipped dependency', tester.bucket_skip)
+
+        return failed_job_containers
+
+    # Iterate over supplied tester data container dict and add edge nodes to the DAG (create dependencies)
+    # Note: This method ignores all errors
+    def createEdgeNodes(self, job_container_dict):
+        for tester_name, job_container in job_container_dict.iteritems():
+            tester = job_container.getTester()
+            job_dag = job_container.getDAG()
+
+            # Attempt to add this node to the DAG (prereq iteration may have already added this node)
+            job_dag.add_node_if_not_exists(job_container)
+
+            # Create the edge dependency DAG. Handle any Cyclic errors as passes (but we set the actual test as a failure).
+            for prereq in tester.getPrereqs():
+
+                # we do not care about missing prereqs
+                if prereq not in job_container_dict.keys():
+                    continue
```

Comment:
This is would cause a KeyError, because the prereq value listed in the 'tests' file is invalid, and was already detected by the checkGroupFailures method ('unknown dependency'). So due to 'prereq' being a bogus value at the moment, I continue to the next iteration.